nachos/src/main/java/com/hootsuite/nachos/chip/UpdateableChipSpan.java

```diff
@@ -0,0 +1,552 @@
+package com.hootsuite.nachos.chip;
+
+import android.content.Context;
+import android.content.res.ColorStateList;
+import android.content.res.Resources;
+import android.graphics.Bitmap;
+import android.graphics.Canvas;
+import android.graphics.Paint;
+import android.graphics.Rect;
+import android.graphics.RectF;
+import android.graphics.drawable.Drawable;
+import android.graphics.drawable.StateListDrawable;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v4.content.ContextCompat;
+import android.text.style.ImageSpan;
+
+import com.hootsuite.nachos.R;
+
+/**
+ * A Span that displays text and an optional icon inside of a material design chip. The chip's dimensions, colors etc. can be extensively customized
+ * through the various setter methods available in this class.
+ *     The basic structure of the chip is the following:
+ * For chips with the icon on right:
+ * <pre>
+ *
+ *                                  (chip vertical spacing / 2)
+ *                  ----------------------------------------------------------
+ *                |                                                            |
+ * (left margin)  |  (padding edge)   text   (padding between image)   icon    |   (right margin)
+ *                |                                                            |
+ *                  ----------------------------------------------------------
+ *                                  (chip vertical spacing / 2)
+ *
+ *      </pre>
+ * For chips with the icon on the left (see {@link #setShowIconOnLeft(boolean)}):
+ * <pre>
+ *
+ *                                  (chip vertical spacing / 2)
+ *                  ----------------------------------------------------------
+ *                |                                                            |
+ * (left margin)  |   icon  (padding between image)   text   (padding edge)    |   (right margin)
+ *                |                                                            |
+ *                  ----------------------------------------------------------
+ *                                  (chip vertical spacing / 2)
+ *     </pre>
+ */
+public class UpdateableChipSpan extends ImageSpan implements Chip {
+
+    private static final float SCALE_PERCENT_OF_CHIP_HEIGHT = 0.70f;
+    private static final boolean ICON_ON_LEFT_DEFAULT = true;
+
+    private int[] mStateSet = new int[]{};
+
+    private String mEllipsis;
+
+    private ColorStateList mDefaultBackgroundColor;
+    private ColorStateList mBackgroundColor;
+    private int mTextColor;
+    private int mIconBackgroundColor;
+
+    private int mTextSize = -1;
+    private int mPaddingEdgePx;
+    private int mPaddingBetweenImagePx;
+    private int mLeftMarginPx;
+    private int mRightMarginPx;
+    private int mMaxAvailableWidth = -1;
+
+    private CharSequence mText;
+    private String mTextToDraw;
+
+    private StateListDrawable mIcon;
```

Comment:
⚠️  `StateListDrawable` is just a subclass of `Drawable`. Why couldn't you achieve this change by keeping this a `Drawable`?